[gtest] Support Production builds

JonWBedard opened the pull request:
#### 4b5332a75668d479c92cdf2b9bd694ce300074a7
<pre>
[gtest] Support Production builds
<a href="https://bugs.webkit.org/show_bug.cgi?id=289621">https://bugs.webkit.org/show_bug.cgi?id=289621</a>
<a href="https://rdar.apple.com/146870475">rdar://146870475</a>

Reviewed by Elliott Williams.

* .submitproject-tools: Add gtest.
* Source/ThirdParty/gtest/xcode/Config/StaticLibraryTarget.xcconfig: Enable installhdrs scripts,
set library install path explicitly.
* Source/ThirdParty/gtest/xcode/gtest.xcodeproj/project.pbxproj: Include all headers in
installhdrs. Make gtest-static the default project.

Canonical link: <a href="https://commits.webkit.org/292109@main">https://commits.webkit.org/292109@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/678c16164ea8d1ca749382867be6c943b3a86c87

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/94982 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/14577 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/4434 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/100011 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/45480 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/97031 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/14866 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/22999 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/72454 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/29753 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/97983 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/11090 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/85764 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/52779 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/10800 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/3487 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/44817 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/81004 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/3583 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/102066 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/22024 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/16079 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/81470 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/22271 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/81780 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/80854 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20198 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/25399 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/2805 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/15266 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/21997 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/21654 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/25129 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/23395 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on .submitproject:
```diff
@@ -9,7 +9,6 @@
     excludedSubdirectoryPatterns = Source/ThirdParty/d3flamegraphjs
     excludedSubdirectoryPatterns = Source/ThirdParty/d3js
     excludedSubdirectoryPatterns = Source/ThirdParty/gmock
-    excludedSubdirectoryPatterns = Source/ThirdParty/gtest
```
Can we avoid submitting it in the normal case? I think that we should try hard.

emw-apple commented on Source/ThirdParty/gtest/xcode/gtest.xcodeproj/project.pbxproj:
```diff
@@ -850,6 +895,30 @@
+			buildSettings = {
+				CODE_SIGN_STYLE = Automatic;
+				PRODUCT_NAME = "$(TARGET_NAME)";
+			};
```
Please remove these project-level overrides—I know Xcode puts them there by default, but they just get in the way when we need to customize things like signing. You can either edit them by hand to make
```
buildSettings = {
};
```
or you can use the Build Settings > Customized filter in Xcode, select all, and press delete.

emw-apple commented on Source/ThirdParty/gtest/xcode/gtest.xcodeproj/project.pbxproj:
Out of curiosity, what's the reason to have an "All" aggregate? Do we actually need to build both the staticlib and the dylib?

JonWBedard replied:
Removing that now (or trying to, at least)

emw-apple commented on Source/ThirdParty/gtest/xcode/Config/StaticLibraryTarget.xcconfig:
```diff
@@ -14,10 +14,13 @@ GCC_DYNAMIC_NO_PIC = NO
 // stripped.
 STRIP_STYLE = debugging
 
+INSTALLHDRS_SCRIPT_PHASE = YES;
```
This line seems unnecessary, as you haven't added any script phases

JonWBedard committed:
[gtest] Support Production builds
https://bugs.webkit.org/show_bug.cgi?id=289621
rdar://146870475

Reviewed by Elliott Williams.

* .submitproject-tools: Add gtest.
* Source/ThirdParty/gtest/xcode/Config/StaticLibraryTarget.xcconfig: Enable installhdrs scripts,
set library install path explicitly.
* Source/ThirdParty/gtest/xcode/gtest.xcodeproj/project.pbxproj: Include all headers in
installhdrs. Make gtest-static the default project.

Canonical link: https://commits.webkit.org/292109@main
